org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/TraceRegion.java

```diff
@@ -45,7 +46,15 @@ public TraceRegion(int myOffset, int myLength, int myLineNumber, int myEndLineNu
  		}
 	}
 	
+	/**
+	 * @deprecated use {link {@link #TraceRegion(int, int, int, int, boolean, Collection, AbstractTraceRegion)} instead.
+	 */
+	@Deprecated
 	public TraceRegion(int myOffset, int myLength, int myLineNumber, int myEndLineNumber, boolean useForDebugging, List<ILocationData> allLocationData, AbstractTraceRegion parent) {
+		this(myOffset, myLength, myLineNumber, myEndLineNumber, useForDebugging, (Collection<ILocationData>) allLocationData, parent);
+	}
+	
+	public TraceRegion(int myOffset, int myLength, int myLineNumber, int myEndLineNumber, boolean useForDebugging, Collection<ILocationData> allLocationData, AbstractTraceRegion parent) {
```

Comment:
The collection type is overly constrained. We only need the upperbound, which will make List<ILocationData> directly assignable. So all the casts are not neccessry (except. the one above because of overload resolution).
I.e. `Collection<? extends ILocationData>`